subsys/net/ip/Kconfig.ipv4

```diff
@@ -46,6 +46,16 @@ config NET_IPV4_ACCEPT_ZERO_BROADCAST
 	  If set, then accept UDP packets destined to non-standard
 	  0.0.0.0 broadcast address as described in RFC 1122 ch. 3.3.6
 
+config NET_IPV4_IGMP
+	bool "Internet Group Management Protocol (IGMP) support"
+	select NET_IPV4_HDR_OPTIONS
+	default y
+	help
+	  The value depends on your network needs. IGMP should normally
+	  be active. Currently we support IGMPv2 and earlier versions.
+	  Requires IPv4 header support because IP Router Alert option
+	  must be sent. See RFC 2236 for details.
```

Comment:
I am a bit concerned about `default y`. Is there any added value if application does not use multicast directly?

Personally I was using Zephyr with cellular modems (using PPP) and WiFi to build IoT devices. In both cases I was more or less constrained in flash size and mostly wanted to reach cloud using TCP (HTTP, MQTT) or UDP (CoAP). So I wonder if IGMP is any added value (from what I just learned, it seems to be useful only for multicast). So I would either prefer it disabled by default to not "bloat" default networking configuration too much or alternatively improve this help message a bit to encourage users to disable it in specific cases (like using only unicast connections). At the end I don't want Zephyr networking to be considered "heavy" by new (or existing) users and expecting users to know when to disable rarely used features (like IGMP?).

Reply:
> I am a bit concerned about `default y`. Is there any added value if application does not use multicast directly?

According to RFC 2236, "It is required to be implemented by all hosts wishing to receive IP multicasts". So we could make it optional if we consider that multicast support needs to be enabled separately. There has not been much complaints about this missing IGMP support so this does not seem to be videly used feature among current Zephyr users. So I am ok if we make this optional.

> it seems to be useful only for multicast)

Absolutely, this is only meant for IPv4 multicast.

> At the end I don't want Zephyr networking to be considered "heavy" by new (or existing) users

I agree.